core/src/main/java/org/opensearch/sql/calcite/type/ExprSqlType.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.calcite.type;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.math.BigDecimal;
+import java.nio.charset.Charset;
+import org.apache.calcite.avatica.util.ByteString;
+import org.apache.calcite.sql.SqlCollation;
+import org.apache.calcite.sql.type.BasicSqlType;
+import org.apache.calcite.sql.type.SqlTypeName;
+import org.locationtech.jts.geom.Geometry;
+import org.opensearch.sql.calcite.utils.OpenSearchTypeFactory;
+import org.opensearch.sql.calcite.utils.OpenSearchTypeFactory.ExprUDT;
+
+public class ExprSqlType extends ExprRelDataType<BasicSqlType> {
```

Comment:
Could you add some comments on `ExprSqlType` and `ExprJavaType` to explain why `ExprDateType extends ExprSqlType` but `ExprIPType extends ExprJavaType`